Sources/ParseSwift/Protocols/ParseObjectMutable.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  ParseObjectMutable.swift
+//  ParseSwift
+//
+//  Created by Damian Van de Kauter on 30/10/2021.
+//  Copyright © 2021 Parse Community. All rights reserved.
+//
+
+import Foundation
+
+/**
+ The `ParseObjectMutable` protocol creates an empty copy of the respective object.
+ This can be used when you only need to update a subset of the fields of an object
+ as oppose to updating every field of an object.
+ Using the mutable copy and updating a subset of the fields reduces the amount of data
+ sent between client and server when using `save` and `saveAll`
+ to update objects.
+ 
+ **Example use case for User:**
+ ````
+ struct User: ParseUser, ParseObjectMutable {
+     //: These are required by `ParseObject`.
+     var objectId: String?
+     var createdAt: Date?
+     var updatedAt: Date?
+     var ACL: ParseACL?
+
+     //: These are required by `ParseUser`.
+     var username: String?
+     var email: String?
+     var emailVerified: Bool?
+     var password: String?
+     var authData: [String: [String: String]?]?
+
+     //: Your custom keys.
+     var customKey: String?
+     var score: GameScore?
+     var targetScore: GameScore?
+     var allScores: [GameScore]?
+ }
+ 
+ var user = User.current?.mutable
+ user?.customKey = "newValue"
+
+ do {
+     try await user?.save()
+ } catch {
+     // Handle error
+ }
+ ````
+ 
+ **Example use case for GameScore:**
```

Comment:
```suggestion
 **Example use case for a general `ParseObject`:**
```

Reply:
Note: you can click `Add suggestion to batch` to let GitHub add the change for you